staging/src/k8s.io/component-base/cli/flag/string_slice_flag.go

```diff
@@ -45,9 +46,11 @@ func (s *StringSlice) String() string {
 }
 
 func (s *StringSlice) Set(val string) error {
-	if s.value == nil || !s.changed {
-		v := make([]string, 0)
-		s.value = &v
+	if s.value == nil {
+		return fmt.Errorf("no target (nil pointer to []string)")
```

Comment:
IMO it's better to make the `NewStringSlice` detect this (and e.g. panic)? This is really a coding error, not a runtime error, you want to detect it as soon as possible. It'd be best if you could make it not compile, but I don't see a good way to do that.

Reply:
```golang
func NewStringSlice(s *[]string) *StringSlice {
	if s == nil {
		panic("no target (nil pointer to []string)")
	}
	return &StringSlice{value: s}
}
```
Is this? This is somewhat inconsistent with the previous code style